### ISSUE TITLE
VPopover is missing `role="dialog"`

### ISSUE THREAD

**Issue by obulat** [WordPress/openverse#2076]
## Description

As an element that appears on top of other content and contains focusable elements, `VPopover` should use `role="dialog"`.

## Additional context

See documentation: 
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role
